tests/Properties.hs

```diff
@@ -1614,6 +1614,36 @@ prop_read_write_file_D x = unsafePerformIO $ do
         (const $ do y <- D.readFile f
                     return (x==y))
 
+prop_hgetline_like_s8_hgetline (LinedASCII filetext) (lineEndIn, lineEndOut) = idempotentIOProperty $ do
+    tid <- myThreadId
+    let f = "qc-test-"++show tid
+    let newlineMode = NewlineMode (if lineEndIn then LF else CRLF) (if lineEndOut then LF else CRLF)
+    bracket_
+        (writeFile f filetext)
```

Comment:
On windows this will take the original randomized text and write every `\n` as a `\r\n`, eliminating any lone `\n`s from the test cases.
This needs to be modified so that `haOutputNL = LF` to make the file store the actual test data.